### PR TITLE
Penalties cleanup

### PR DESCRIPTION
**Pull request by jrhea**
## PR Description
- added missing case 2 penalty step
- moved lambdas to class level scope for readability and potential reuse in the future.

### CODE REVIEW

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java [hunk 1]
```diff
@@ -203,58 +203,117 @@ public static void justificationAndFinalization(BeaconState state) throws Except
...
-            }
+      Predicate<Integer> slashed_filter =
+          index -> {
+            Validator validator = state.getValidator_registry().get(index);
+            return validator.getSlashed_epoch().compareTo(BeaconStateUtil.get_current_epoch(state))
+                <= 0;
           };
 
       // prev epoch justified attester
       List<Integer> validator_indices =
```
Comment (schroedingerscode) [inline ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java:240]: Is the below logic stubbed for something else? It looks like validator_indices is just being set to different references without anything actually being performed.

**Review by schroedingerscode — COMMENTED** [review]
One comment about logic.

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java (hunk 1 above)
Comment (jrhea) [reply ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java:240]: It's a lambda function that collection.filter uses

**Review by schroedingerscode — APPROVED** [review]
LGTM

**Pull request merged**